test/session_test.rb

```diff
@@ -84,6 +117,36 @@ def setup
     end
   end
 
+  test "provides default nil access_scopes attribute" do
+    session = ShopifyAPI::Session.new(
+      domain: "testshop.myshopify.com",
+      token: "any-token",
+      api_version: any_api_version
+    )
+    assert_nil session.access_scopes
+  end
+
+  test "provides specified nil access_scopes attribute" do
+    session = ShopifyAPI::Session.new(
+      domain: "testshop.myshopify.com",
+      token: "any-token",
+      access_scopes: "read_products",
+      api_version: any_api_version
+    )
+    assert_equal "read_products", session.access_scopes.to_s
+  end
+
+  test "session instantiation raises error if bad access scopes are provided" do
+    assert_raises do
```

Comment:
What does it raise?

Reply:
Currently it's `NoMethodError` because the `strip` method doesn't exist on the object passed in object passed in. I can specify `NoMethodError`

Reply:
Should we have a custom error for `ApiAccess` to throw when an invalid input is passed in?

Reply:
I think that because specifying `api_access` is opt-in, it's up to the user to specify good input. In this case, building a new session will either:

* parse access_scopes correctly or
* throw some error (automatically an invalid session)

For this test, we know that it's `NoMethodError` because `strip` doesn't exist on the hash, but really the bad input could throw anything.